shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java

```diff
@@ -52,101 +55,156 @@
 public class PlatformPluginTest {
   private final Context ctx = ApplicationProvider.getApplicationContext();
 
-  @Config(sdk = Build.VERSION_CODES.KITKAT)
-  @Test
-  public void itIgnoresNewHapticEventsOnOldAndroidPlatforms() {
-    View fakeDecorView = mock(View.class);
-    Window fakeWindow = mock(Window.class);
+  private View fakeDecorView = mock(View.class);
+  private Window fakeWindow = mock(Window.class);
+  private Activity fakeActivity = mock(Activity.class);
+  private PlatformChannel mockPlatformChannel = mock(PlatformChannel.class);
+  private PlatformPluginDelegate mockPlatformPluginDelegate = mock(PlatformPluginDelegate.class);
+  private PlatformPlugin testPlatformPlugin = new PlatformPlugin(fakeActivity, mockPlatformChannel);
+  private PlatformPlugin testPlatformPluginWithDelegate =
+      new PlatformPlugin(fakeActivity, mockPlatformChannel, mockPlatformPluginDelegate);
+
+  private ClipboardManager clipboardManager;
+  private ClipboardContentFormat clipboardFormat;
```

Comment:
When possible, it's better to have stand-alone setup methods that return objects, rather than making fixtures stateful; I really like https://abseil.io/tips/122 on this topic (written for C++, but applicable to any language).

Maybe that's not possible here; it looks like a fair number of these variables are individually referenced in tests rather than only being used internally. When it's not I would actually prefer duplication in tests rather than a lot of fixture state, but I don't feel too strongly about it here so if you and @reidbaker prefer this, that's fine with me.

The one exception would be the `testPlatformPlugin` and `testPlatformPluginWithDelegate`. I feel pretty strongly that the object under test itself should not be part of the fixture whenever possible, and the fact that there are different versions here, each of which only apply to some tests, is even more indication that these aren't really part of the overall test state, but part of tests. If you want to abstract out the creation, you could always make a stand-alone getter instead. E.g.,:

```java
private PlatformPlugin createTestPlugin(bool withDelegate) {
  if (withDelegate) {
    return new PlatformPlugin(fakeActivity, mockPlatformChannel, mockPlatformPluginDelegate);
  } else {
    return new PlatformPlugin(fakeActivity, mockPlatformChannel);
  }
}
```

(Or two different methods, one for delegate, one for no delegate)

Reply:
Ah okay I was wondering if I went overboard :) thanks for the tips and explanation! This was good to learn.

I ended up keeping the setup method for clipboard tests (the ones related to this PR) and getting rid of the rest of the changes I made, leaving most of the duplication which is likely safer as you described.